### PR TITLE
Introduce global `currentUser` object to JS environment

### PR DESCRIPTION
**Pull request by vast**
Instead of relying on something visible only for authenticated user in DOM (which is ugly and fragile):

```coffee
if $('.current_user:first').size()
  @sendRating(value)
```

we should hide implementation details and use global `currentUser` object. `currentUser` is set up from JSON data in application layout. With it example above would look like:

```coffee
if App.currentUser.isSignedIn()
  @sendRating(value)
```

This PR also shows how to properly pass any global backend variables to client-side. For example, [flipper](https://github.com/jnunemaker/flipper) feature toggles could look like:

```slim
javascript:
  App.currentUserData = #{raw current_user.to_json};
  App.$flipper = #{raw js_feature_toggles };
```

```ruby
def js_feature_toggles
  features_state = $flipper.features.map do |feature|
    [feature.name, $flipper[feature.name].enabled?]
  end

  Hash[features_state].to_json
end
```

```coffee
# initialize ckeditor support only if this feature is toggled on
if App.$flipper.ckeditor 
  @$("textarea").ckeditor()
```

### CODE REVIEW

File: app/assets/javascripts/current_user.coffee [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class User
+  constructor: (@data = {}) ->
+
+  get: (param) ->
+    @data[param]
+
+  isSignedIn: ->
+    @get("id")?
+
+@currentUser = App.currentUser = new User(App.currentUserData)
```
Comment (timurvafin) [inline app/assets/javascripts/current_user.coffee]: Could you please clarify what is the preferred what to access `currentUser` -- as global variable or from `App.currentUser`?

Comment (vast) [inline app/assets/javascripts/current_user.coffee]: I personally don't like having global variables in JS: it's easy to spoil it by forgetting to put `var` before. But it's definitely much shorter than `App.currentUser` and I saw a lot of people using it like that.

If you also don't like global variables let's leave only `App.currentUser`. Developers can make it global in any time.

Comment (timurvafin) [inline app/assets/javascripts/current_user.coffee]: :+1:  for `App.currentUser`

**Comment by vast** [comment]
@timurvafin, thanks, removed `window.currentUser`

**Comment by timurvafin** [comment]
:ok:

**Pull request merged**